core/templates/dev/head/pages/exploration_player/HintAndSolutionModalService.js

```diff
@@ -84,13 +88,17 @@ oppia.factory('HintAndSolutionModalService', [
                 interaction);
               $scope.solutionExplanationHtml =
                 solution.getOppiaSolutionExplanationResponseHtml();
+
               $scope.closeModal = function() {
+                $uibModalInstance.dismiss('cancel');
+              };
+
+              $scope.$on('modal.closing', function() {
```

Comment:
Is there any reason behind handling this in a different way? (as a listener) Can we handle this in a similar way?

Reply:
I think that unintentionally I introduced here a problem because I don't check if the modal is closing because of being dismissed or closed.
I did it in another way mostly because _HintAndSolutionModalService_ is a factory of modals and in order to reduce the code I handle dismiss in the produced modal and not in every place that this factory is used.